src/AkamaiContentControlClient.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @file
+ * Contains Drupal\akamai\AkamaiContentControlClient.
+ */
+
+namespace Drupal\akamai;
+
+use Drupal\Core\Http\ClientFactory;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Exception\ClientException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Provides a service to interact with the Akamai Content Control REST API.
+ */
+class AkamaiContentControlClient implements AkamaiContentControl {
+
+  /**
+   * The HTTP client to fetch the feed data with.
+   *
+   * @var \Drupal\Core\Http\ClientFactory
+   */
+  protected $httpClientFactory;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  // A config object.
+  protected $config;
+
+  /**
+   * Constructs an AkamaiContentControlClient object.
+   *
+   * @todo Inject logger
+   * @todo Inject HTTPClient
+   * @todo Inject config
+   */
+  public function __construct() {
+    // @todo Inject the services.
+    $this->httpClient = \Drupal::httpClient();
+    $this->logger = \Drupal::logger('akamai');
```

Comment:
@cam8001, I recently figured out dependency injection myself.   Services like logger should be injected.